src/step-12-database/test/unit/login.test.js

```diff
@@ -109,4 +109,27 @@ test('POST /login', async t => {
     t.equal(res.statusCode, 200)
     t.equal((await res.json()).token, 'jwt token')
   })
+
+  t.test('stores the signed JWT', async t => {
+    const fastify = buildServer()
+
+    fastify.pg.query.resolves({
+      rows: [{ id: 1, username: 'alice' }],
+    })
+    fastify.jwt.sign.returns('jwt token')
+
+    await fastify.inject({
+      url: '/login',
+      method: 'POST',
+      body: {
+        username: 'alice',
+        password: 'alice',
+      },
+    })
+
+    t.equal(fastify.jwt.sign.callCount, 1)
+    t.same(fastify.jwt.sign.getCall(0).args[0], {
+      username: 'alice'
+    })
```

Comment:
minor: let's prefer a more readable way to make these assertions. check out https://sinonjs.org/releases/latest/assertions/

Reply:
Ah nice, I've not used tap before so was wondering how to get the sinon-chai `.to.have.been.calledwith()` type assertions.

Reply:
you can simply use sinon assertions. if they fail they will just result in an exception and the tap test will fail. tap works with any assertion library basically, you don't have to use the built-in assertions (which btw are quite limited)